CI: Test_clientMultipleAPIServers failure

```
--- FAIL: Test_clientMultipleAPIServers (0.02s)

    logger.go:256: time=2025-05-08T09:29:25.742Z level=INFO source=/home/runner/work/cilium/cilium/pkg/k8s/client/restConfig_provider.go:274 msg="Rotated api server" module=k8s-client url=http://127.0.0.1:33993/
    logger.go:256: time=2025-05-08T09:29:25.746Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:357 msg="Starting hive"
    logger.go:256: time=2025-05-08T09:29:25.746Z level=INFO source=/home/runner/work/cilium/cilium/pkg/k8s/client/cell.go:317 msg="Establishing connection to apiserver" module=k8s-client ipAddr=http://127.0.0.1:33993/
    logger.go:256: time=2025-05-08T09:29:25.748Z level=INFO source=/home/runner/work/cilium/cilium/pkg/k8s/client/cell.go:344 msg="Connected to apiserver" module=k8s-client
    logger.go:256: time=2025-05-08T09:29:25.749Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:361 msg="Started hive" duration=3.055708ms
    logger.go:256: time=2025-05-08T09:29:25.756Z level=WARN source=/home/runner/work/cilium/cilium/pkg/k8s/client/cell.go:399 msg="Heartbeat timed out, restarting client connections" module=k8s-client
    logger.go:256: time=2025-05-08T09:29:25.760Z level=INFO source=/home/runner/work/cilium/cilium/pkg/k8s/client/restConfig_provider.go:274 msg="Rotated api server" module=k8s-client url=http://127.0.0.1:40725/
    client_test.go:337: 
        	Error Trace:	/home/runner/work/cilium/cilium/pkg/k8s/client/client_test.go:337
        	Error:      	Received unexpected error:
        	            	Get "http://127.0.0.1:40725/api/v1/namespaces/test/pods/slim-pod": dial tcp 127.0.0.1:40725: connect: connection refused
        	Test:       	Test_clientMultipleAPIServers
```

https://github.com/cilium/cilium/actions/runs/14903032174/job/41859123372

Seen on https://github.com/cilium/cilium/pull/39426

/cc @cilium/sig-k8s @aditighag

--- a/pkg/k8s/client/client_test.go
+++ b/pkg/k8s/client/client_test.go
@@ -227,7 +227,9 @@ func Test_client(t *testing.T) {
 	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
 	hive.RegisterFlags(flags)
 	flags.Set(option.K8sAPIServerURLs, srv.URL)
-	flags.Set(option.K8sHeartbeatTimeout, "5ms")
+	flags.Set(option.K8sHeartbeatTimeout, "150ms")
+	// Set a higher QPS limit as the test exercises timing aspects.
+	flags.Set(option.K8sClientQPSLimit, "500")
 
 	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
 	defer cancel()
@@ -311,7 +313,9 @@ func Test_clientMultipleAPIServers(t *testing.T) {
 	hive.RegisterFlags(flags)
 	urls := []string{servers[0].URL, servers[1].URL, servers[2].URL}
 	flags.Set(option.K8sAPIServerURLs, strings.Join(urls, ","))
-	flags.Set(option.K8sHeartbeatTimeout, "5ms")
+	flags.Set(option.K8sHeartbeatTimeout, "150ms")
+	// Set a higher QPS limit as the test exercises timing aspects.
+	flags.Set(option.K8sClientQPSLimit, "500")
 	// 2/3 servers are stopped in order to validate that the agent connects to an active server.
 	servers[1].Close()
 	servers[2].Close()

